NDTensors/src/blocksparse/linearalgebra.jl

```diff
@@ -296,6 +296,71 @@ function LinearAlgebra.eigen(
   return D, V, Spectrum(d, truncerr)
 end
 
+ql(T::BlockSparseTensor{ElT,2}; kwargs...) where {ElT} = qx(ql, T; kwargs...)
```

Comment:
Let's use the style:
```julia
ql(T::BlockSparseTensor{<:Any,2}; kwargs...) = qx(ql, T; kwargs...)
```
Basically, a "rule" would be try to avoid ` where {...}` unless it is unavoidable, i.e. you actually use the type and the type isn't easy to access through a function. For example, even if we were using `ElT`, you can easily access that same information with the function `eltype(T)`.

We have been moving towards that style, which makes the code cleaner and makes it easier to write generic code.

Reply:
Please make the same changes to the rest of the code involved in this PR, such as the dense and block sparse `qr`, `qx`, etc.

Reply:
OK I think I got them all for Elt.   We still have where {IndsT} in the dense qx() function.  Is it better to use
indstype(::Type{<:Tensor{<:Any,<:Any,<:Any,IndsT}}) where {IndsT} = IndsT
indstype(T::Tensor) = indstype(typeof(T))

for this as well?

Reply:
I tried it and it seems to be introducing subtle bugs, I am probably not doing it right.

Reply:
I left some other comments on other cases of ` where ` that I think you can remove. Please post in those comments what you tried and how it didn't work if you see issues.

Reply:
Fixed